Fully resetting hystrix during testing to avoid exceptions

I'm using my own metrics publisher that I've registered.  When running tests, exactly one of them will succeed, and all of the others will fail due to the following:

     Caused by: java.lang.IllegalStateException: Another strategy was already registered.
        at com.netflix.hystrix.strategy.HystrixPlugins.registerMetricsPublisher(HystrixPlugins.java:180) ~[hystrix-core-1.3.18.jar:na]

In googling, I found other people with the same issue: https://groups.google.com/forum/#!topic/hystrixoss/Z7NKlGMr1PA

In that topic, they suggest the following.  It works, and I realize I don't need it all in my particular case.  But would it be possible to encapsulate all of the following into a single helper method to fully reset the hystrix state between tests?

    Hystrix.reset() 
    HystrixPlugins.getInstance().@concurrencyStrategy.set(null);
    HystrixPlugins.getInstance().@metricsPublisher.set(null);
    HystrixPlugins.getInstance().@notifier.set(null);
    HystrixPlugins.getInstance().@propertiesFactory.set(null);
    HystrixPlugins.getInstance().@commandExecutionHook.set(null);

--- a/hystrix-core/src/main/java/com/netflix/hystrix/Hystrix.java
+++ b/hystrix-core/src/main/java/com/netflix/hystrix/Hystrix.java
@@ -4,6 +4,7 @@ import java.util.LinkedList;
 import java.util.NoSuchElementException;
 import java.util.concurrent.TimeUnit;
 
+import com.netflix.hystrix.strategy.HystrixPlugins;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -56,6 +57,7 @@ public class Hystrix {
         HystrixCollapser.reset();
         // clear circuit breakers
         HystrixCircuitBreaker.Factory.reset();
+        HystrixPlugins.reset();
     }
 
     private static ThreadLocal<LinkedList<HystrixCommandKey>> currentCommand = new ThreadLocal<LinkedList<HystrixCommandKey>>() {

--- a/hystrix-core/src/main/java/com/netflix/hystrix/strategy/HystrixPlugins.java
+++ b/hystrix-core/src/main/java/com/netflix/hystrix/strategy/HystrixPlugins.java
@@ -56,6 +56,17 @@ public class HystrixPlugins {
     }
 
     /**
+     * Reset all of the HystrixPlugins to null.  You may invoke this directly, or it also gets invoked via <code>Hystrix.reset()</code>
+     */
+    public static void reset() {
+        getInstance().notifier.set(null);
+        getInstance().concurrencyStrategy.set(null);
+        getInstance().metricsPublisher.set(null);
+        getInstance().propertiesFactory.set(null);
+        getInstance().commandExecutionHook.set(null);
+    }
+
+    /**
      * Retrieve instance of {@link HystrixEventNotifier} to use based on order of precedence as defined in {@link HystrixPlugins} class header.
      * <p>
      * Override default by using {@link #registerEventNotifier(HystrixEventNotifier)} or setting property: <code>hystrix.plugin.HystrixEventNotifier.implementation</code> with the full classname to
@@ -278,5 +289,6 @@ public class HystrixPlugins {
             return null;
         }
     }
+
 
 }

